### PR TITLE
Commit generated openapi definitions

### PR DESCRIPTION
**Pull request by dims**
Check in pkg/generated/openapi/zz_generated.openapi.go

This will help extract say hyperkube into its own repository in the long
run.

Doing the same thing as was done for apiextensions in https://github.com/kubernetes/kubernetes/pull/75339

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -119,6 +119,8 @@ zz_generated_*_test.go
 !staging/src/k8s.io/code-generator/_examples/apiserver/openapi/zz_generated.openapi.go
 # low-change sample-apiserver spec to be compilable when published
 !staging/src/k8s.io/sample-apiserver/pkg/generated/openapi/zz_generated.openapi.go
+# TODO(dims): remove this when we stop committing the generated definition
```
Comment (liggitt) [inline .gitignore]: adding  this while also adding a TODO to remove it is confusing

**Comment by liggitt** [comment]
is there a verification script that ensures the committed content stays up to date?

**Comment by liggitt** [comment]
> Doing the same thing as was done for apiextensions in #75339

The reason for that was to ensure all published staging repos can be built as-is. There is no such statement for k/k. If we intend for k/k to be vendorable/buildable, shouldn't there be a test ensuring that?

**Comment by dims** [comment]
@liggitt the existing verify scripts catch this already, example see https://github.com/kubernetes/kubernetes/pull/81239/commits/9d83d479d412e3a7682206efeaabf68a9bc6461f which got caught

**Comment by BenTheElder** [comment]
Does this mean we intend to support importing our commands from this repo ...?

**Comment by dims** [comment]
@roycaihw WDYT?

**Review by feiskyer — COMMENTED** [review]
lgtm

**Comment by roycaihw** [comment]
my understanding is that we don't commit this generated file because

1. it caused merge conflicts in the past: https://github.com/kubernetes/kubernetes/pull/41106
2. k/k isn't claimed to be vendorable/buildable so far

#75339 was not ideal, but got committed because it didn't fall under those two points.

**Pull request closed without merging**